### PR TITLE
add spookyswap fantom to dex trades - Ready for Review

### PR DESCRIPTION
**Pull request by henrystats**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @henrystats 

This PR adds `spookyswap_fantom_trades` to `dex_trades`.

- Metadata & naming conventions ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercase ✅
- Inner join to large tables(fantom.transactions) as possible ✅

[Sample spookyswap fantom trades](https://dune.com/queries/1948440)

**Comment by henrystats** [comment]
Thank you soo much ser 🫡 for all the reviews!

File: models/spookyswap/fantom/spookyswap_fantom_trades.sql [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        case when amount0In = 0 then f.token1 else f.token0 end as token_sold_address,
+        t.contract_address as project_contract_address,
+        t.evt_tx_hash as tx_hash,
+        '' as trace_address,
+        t.evt_index
+    FROM
+        {{ source('spookyswap_fantom', 'Pair_evt_Swap') }} t
+        inner join {{ source('spookyswap_fantom', 'UniswapV2Factory_evt_PairCreated') }} f 
+            on f.pair = t.contract_address
+    {% if not is_incremental() %}
```
Comment (jeff-dude) [inline models/spookyswap/fantom/spookyswap_fantom_trades.sql:36]: are we missing the is_incremental logic?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/spookyswap/fantom/spookyswap_fantom_trades.sql (hunk 1 above)
Comment (jeff-dude) [reply models/spookyswap/fantom/spookyswap_fantom_trades.sql:36]: also please do fix the merge conflicts by merging main into your branch

Comment (henrystats) [reply models/spookyswap/fantom/spookyswap_fantom_trades.sql]: fixed incremental and merged conflicts ser @jeff-dude

**Comment by jeff-dude** [comment]
will rerun CI test once prod is caught up on current refresh of dex trades

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**